src/processor.ts

```diff
@@ -1,17 +1,45 @@
 import {
-  FirehoseTransformationHandler,
-  FirehoseTransformationResultRecord,
+  FirehoseTransformationHandler as Handler,
+  FirehoseTransformationEventRecord as EventRecord,
+  FirehoseTransformationResultRecord as ResultRecord,
 } from "aws-lambda";
+import { DateTime } from "luxon";
+import { create } from "superstruct";
+import { OCSEvent } from "./processor/structs";
 
-const handler: FirehoseTransformationHandler = async ({ records }) => ({
-  records: records.map(
-    ({ recordId, data }): FirehoseTransformationResultRecord => ({
+export const handler: Handler = async ({ records }) => ({
+  records: records.map(transformRecord),
+});
+
+const transformRecord = ({ recordId, data }: EventRecord): ResultRecord => {
+  try {
+    const event = JSON.parse(Buffer.from(data, "base64").toString());
+    const { time, data: { raw } } = create(event, OCSEvent); // prettier-ignore
+
+    return {
       recordId,
-      data,
       result: "Ok",
-      metadata: { partitionKeys: { serviceDay: "1970-01-01" } },
-    })
-  ),
-});
+      data: raw,
+      metadata: { partitionKeys: { serviceDay: serviceDay(time) } },
+    };
+  } catch (error) {
+    return { recordId, result: "ProcessingFailed", data };
```

Comment:
When we add Sentry, I'd like to also send a report here, as otherwise we don't know what the error was (any returned `data` is ignored — the field is actually optional if the result is `ProcessingFailed`, but the TypeScript types don't account for this).